app/src/main/java/io/homeassistant/companion/android/notifications/MessagingManager.kt

```diff
@@ -294,271 +294,238 @@ class MessagingManager @Inject constructor(
         } ?: ServerManager.SERVER_ID_ACTIVE
         jsonData = jsonData + mutableMapOf<String, String>().apply { put(THIS_SERVER_ID, serverId.toString()) }
 
-        when {
-            jsonData[NotificationData.MESSAGE] == REQUEST_LOCATION_UPDATE -> {
-                Log.d(TAG, "Request location update")
-                requestAccurateLocationUpdate()
-            }
-            jsonData[NotificationData.MESSAGE] == CLEAR_NOTIFICATION && !jsonData["tag"].isNullOrBlank() -> {
-                Log.d(TAG, "Clearing notification with tag: ${jsonData["tag"]}")
-                clearNotification(jsonData["tag"]!!)
-            }
-            jsonData[NotificationData.MESSAGE] == REMOVE_CHANNEL && !jsonData[NotificationData.CHANNEL].isNullOrBlank() -> {
-                Log.d(TAG, "Removing Notification channel ${jsonData[NotificationData.CHANNEL]}")
-                removeNotificationChannel(jsonData[NotificationData.CHANNEL]!!)
-            }
-            jsonData[NotificationData.MESSAGE] == TextToSpeechData.TTS -> {
-                speakText(context, jsonData)
-            }
-            jsonData[NotificationData.MESSAGE] in DEVICE_COMMANDS -> {
-                Log.d(TAG, "Processing device command")
-                when (jsonData[NotificationData.MESSAGE]) {
-                    COMMAND_DND -> {
-                        if (jsonData[NotificationData.COMMAND] in DND_COMMANDS) {
-                            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
-                                handleDeviceCommands(jsonData)
-                            } else {
-                                mainScope.launch {
+        mainScope.launch {
+            val allowCommands = serverManager.integrationRepository(serverId).isTrusted()
+            when {
+                jsonData[NotificationData.MESSAGE] == REQUEST_LOCATION_UPDATE && allowCommands -> {
+                    Log.d(TAG, "Request location update")
+                    requestAccurateLocationUpdate()
+                }
+                jsonData[NotificationData.MESSAGE] == CLEAR_NOTIFICATION && !jsonData["tag"].isNullOrBlank() && allowCommands -> {
```

Comment:
Out of all the commands this one I feel should be left out as a lot of users make use of this to synchronize things like acknowledgement or to not notify if a certain situation is no longer occurring. 

If we were going to disallow this then we may consider also stopping replacing a notification.

Reply:
I simply followed the list on https://companion.home-assistant.io/docs/notifications/notification-commands, but also agree that notification commands for notifications are probably fine. Where do we draw the line / how to communicate this clearly?

Reply:
So allow `clear_notification` (because you can update) and `remove_channel` (because you can create new channels with settings)?

Reply:
Yea `channel`s are not really removed either, more like a soft delete plus we don't block creation of them so why block clearing the list? Its not like they won't get recreated again 🤷‍♂️

Reply:
Excluded + opened docs PR to document this